bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/device/TapoBridgeHandler.java

```diff
@@ -0,0 +1,174 @@
+        this.config = getThing().getConfiguration();
+        this.cloudConnector = new TapoCloudConnector(getThing(), httpClient);
+        this.uid = getThing().getUID().toString();
+    }
+
+    @Override
+    /**
+     * INIT BRIDGE
+     * set credentials and login cloud
+     */
+    public void initialize() {
+        logger.debug("Initializing Bridge");
+        String username = "";
+        String password = "";
+
+        try {
+            this.config = getThing().getConfiguration();
+            username = config.get(CONFIG_EMAIL).toString();
+            password = config.get(CONFIG_PASS).toString();
+        } catch (Exception e) {
+            logger.error("{} Bridge configuration error: '{}'", this.uid, e.toString());
+        }
+        loginCloud(username, password);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("{} Bridge doesn't handle command: {}", this.uid, command);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("{} dispose bridge", this.uid);
+        stopScheduler();
+        updateStatus(ThingStatus.OFFLINE);
+        super.dispose();
+    }
+
+    /**
+     * Start scheduler
+     */
+    protected void startScheduler() {
+        logger.debug("{} starting bridge sheduler", this.uid);
+        String interval = getThing().getConfiguration().get(CONFIG_CLOUD_UPDATE_INTERVAL).toString();
+        Integer pollingInterval = Integer.valueOf(interval);
+
+        this.pollingJob = scheduler.scheduleWithFixedDelay(this::schedulerAction, 0, pollingInterval, TimeUnit.HOURS);
+    }
+
+    /**
+     * Stop scheduler
+     */
+    protected void stopScheduler() {
+        logger.debug("{} stopping bridge sheduler", this.uid);
+        if (this.pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+    }
+
+    /**
+     * Scheduler Action
+     */
+    protected void schedulerAction() {
+        loginCloud();
+    }
+
+    /**
+     * 
+     * @param username
+     * @param password
+     * @return
+     */
+    public Boolean loginCloud(String username, String password) {
+        credentials.setCredectials(username, password);
+        return loginCloud();
+    }
+
+    /**
+     * Login to Cloud
+     * 
+     * @return
+     */
+    public boolean loginCloud() {
+        String username = credentials.getUsername();
+        String password = credentials.getPassword();
+        if (username != "" && password != "") {
+            logger.debug("{} login with user {}", this.uid, username);
+            if (cloudConnector.login(username, password)) {
+                logger.debug("login success");
+                updateStatus(ThingStatus.ONLINE);
+                return true;
+            } else {
+                logger.debug("login failed");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, cloudConnector.errorMessage());
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Please check all.